core/arch/aarch64/clean_call_opt.c

```diff
@@ -387,21 +448,100 @@ void
 insert_inline_reg_save(dcontext_t *dcontext, clean_call_info_t *cci,
                        instrlist_t *ilist, instr_t *where, opnd_t *args)
 {
-    ASSERT_NOT_IMPLEMENTED(false); /* FIXME i#1569: NYI on AArch64 */
+    callee_info_t *ci = cci->callee_info;
+    int i;
+    /* Don't spill anything if we don't have to. */
+    if (cci->num_regs_skip == NUM_GP_REGS && cci->skip_save_flags &&
+        !ci->has_locals) {
+        return;
+    }
+    /* Spill a register to TLS and point it at our unprotected_context_t.*/
+    PRE(ilist, where, instr_create_save_to_tls
+        (dcontext, ci->spill_reg, TLS_REG2_SLOT));
+    insert_get_mcontext_base(dcontext, ilist, where, ci->spill_reg);
+    /* We use stp to spill consecutive registers and str to spill a single reg
+     * XXX remove duplication */
```

Comment:
I don't understand the comment: what duplication?

Reply:
Those comments do not apply any more, I removed them.